### DISCUSSION TITLE
How To Monitor TeslaMate Status?

### DISCUSSION THREAD

**Discussion by tydalforce** [teslamate-org/teslamate#4374] [Q&A]
I'm trying to find a way to monitor the status of TeslaMate itself.  I didn't find anything in the docs or discussion threads (unless I'm looking in the wrong place). 
 
I just logged into mine a little bit ago and realized the authentication to Tesla's servers expired and I had to create new keys -- it happened a month ago, so I lost a month's worth of data (I check this thing randomly)
I have monitoring in place to tell if TeslaMate is running or not, but it's basically just verifying a response to HTTP requests.  Is there any way to monitor for problems like expired authentication?  A status API Endpoint or something?  I don't need much detail here, just a way of automated monitoring to know "something's wrong" and let me know I need to look at it.

**Comment by tydalforce** [comment]
I may have figured this out, at least good enough for my purposes --- looks like the MQTT topic "teslamate/cars/1/healthy" shows true/false when it's working/not; I'm not clear how this might differ if I had multiple cars (would I have to monitor each?) but I only have one so it's good enough for me (for now).  Uptime Kuma supports MQTT topics, so yay - I think I have monitoring in place now.

I'm def curious if anyone has a better way or whatever (c:

**Reply by brianmay** [reply]
I think that would mark teslamate is unhealthy if the car is unhealthy for any reason. Possibly not ideal?

**Reply by tydalforce** [reply]
I suppose that depends what else could make that return false?  The MQTT page in the documentation (https://docs.teslamate.org/docs/integrations/mqtt) describes this as: 
"Health status of the logger for that vehicle"
I could see it turning false for other reasons, but maybe those reasons are important enough I'd want an alert anyway?

**Comment by brianmay** [comment]
I was just thinking teslamate needed health check endpoints, that ensures the database is online, etc.

Here is a blog post on endpoints for kubernetes: https://www.cogini.com/blog/kubernetes-health-checks-for-elixir-apps/

Here is a change I made to a similar project: https://github.com/brianmay/phone_db/commit/25c1eb935fe69067119c6ed25d147e5cdbe87709:

A lot of this is not needed in teslamate because it already has these things, mostly look at the *.ex files.

I did not look at two potential issues:

* I don't think we redirect http to https but if we did that (now or in the future) it could be a problem if accessing the health check from http not https.
* I didn't even attempt to try to stop logging of health check requests. Which could be numerous depending on how you use it.

**Reply by tydalforce** [reply]
I don't know that a health check would need to be too fancy; once alerted that there's a problem one can go take a look at it and hopefully troubleshoot the problem based on error messages and whatnot.   Maybe something that says "OK" when all is well, and "ERROR" otherwise - or even a simple error message like "Cannot connect to DB" 
I don't know how anyone else uses theirs or how often they go look at the data - I tend to ignore it for large chunks of time.
